kernel/src/process.rs

```diff
@@ -957,6 +977,23 @@ pub struct FunctionCall {
     pub pc: usize,
 }
 
+/// This is similar to `FunctionCall` but for the special case of the Null
+/// Upcall for a subscribe. Because there is no function pointer in a Null
+/// Upcall we can only return these values to userspace. This is used to pass
+/// around upcall parameters when there is no associated upcall to actually call
+/// or userdata for arg3.
```

Comment:
```suggestion
/// or userdata for argument2.
```
?, matching the actual struct below